sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/common/exception/StatusRuntimeExceptionMapper.java

```diff
@@ -22,22 +22,80 @@
 import io.stargate.sgv2.docsapi.api.common.exception.model.dto.ApiError;
 import javax.ws.rs.core.Response;
 import org.jboss.resteasy.reactive.RestResponse;
+import org.jboss.resteasy.reactive.RestResponse.ResponseBuilder;
 import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
 
 /** Simple exception mapper for the {@link StatusRuntimeException}. */
 public class StatusRuntimeExceptionMapper {
 
   @ServerExceptionMapper
-  public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException exception) {
+  public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException grpcE) {
     // TODO finalize mapper, messages, logs, etc.
 
-    Status status = exception.getStatus();
-    ApiError body = new ApiError(status.getDescription());
+    Status.Code sc = grpcE.getStatus().getCode();
 
-    return switch (status.getCode()) {
-      case UNAVAILABLE -> RestResponse.status(Response.Status.BAD_GATEWAY, body);
-      case UNAUTHENTICATED -> RestResponse.status(Response.Status.UNAUTHORIZED, body);
-      default -> RestResponse.status(Response.Status.INTERNAL_SERVER_ERROR, body);
-    };
+    // Handling partly based on information from:
+    // https://developers.google.com/maps-booking/reference/grpc-api/status_codes
+    switch (sc) {
```

Comment:
This `switch` does nothing as you are not returning anything.. This effectively always ends up going to the line `91`. 

See above should be:

```
return swicth() {
}
```

Reply:
Instead of `return switch...`, each case is doing the `return`, so I think this is working

Reply:
ok could be actually yea.. just extend the tests then, sorry for confusion..